core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -112,6 +112,11 @@ private object GroupMetadata {
       Stable -> Set(CompletingRebalance),
       PreparingRebalance -> Set(Stable, CompletingRebalance, Empty),
       Empty -> Set(PreparingRebalance))
+  val unknownMemberMetadata = new MemberMetadata("", "", "", "", 0, 0, "", List[(String, Array[Byte])]())
```

Comment:
Hmm.. I think it would be clearer to track the new members in a separate set (say `pendingMembers`). It's a bit dangerous to mix int the pending members with sentinel metadata in the main collection. Consider, for example, the check below `hasAllMembersJoined` which compares the number of members (which may include some unknown) with the number that are awaiting join. That check no longer seems to work.

Reply:
Yep make sense, I'm on it.